itests/src/test/java/org/apache/unomi/itests/SegmentIT.java

```diff
@@ -54,4 +58,43 @@ public void testSegments() {
         Assert.assertEquals("Segment metadata list should be empty", 0, segmentMetadatas.size());
         LOGGER.info("Retrieved " + segmentMetadatas.size() + " segment metadata entries");
     }
+
+    @Test(expected = BadSegmentConditionException.class)
+    public void testSegmentWithNullCondition() {
+        Metadata segmentMetadata = new Metadata(SEGMENT_ID);
+        Segment segment = new Segment();
+        segment.setMetadata(segmentMetadata);
+        segment.setCondition(null);
+
+        segmentService.setSegmentDefinition(segment);
+    }
+
+    @Test(expected = BadSegmentConditionException.class)
+    public void testSegmentWithInValidCondition() {
+        Metadata segmentMetadata = new Metadata(SEGMENT_ID);
+        Segment segment = new Segment();
+        segment.setMetadata(segmentMetadata);
+        Condition condition = new Condition();
+        condition.setParameter("param", "param value");
+        condition.setConditionTypeId("fakeConditionId");
+        segment.setCondition(condition);
+
+        segmentService.setSegmentDefinition(segment);
+    }
+
+    @Test
+    public void testSegmentWithValidCondition() {
+        Metadata segmentMetadata = new Metadata(SEGMENT_ID);
+        Segment segment = new Segment(segmentMetadata);
+        Condition segmentCondition = new Condition(definitionsService.getConditionType("pastEventCondition"));
+        segmentCondition.setParameter("minimumEventCount", 2);
```

Comment:
btw, we had errors where numberOfDays or minimumEventCount were given string instead of int.
maybe this can also be tested

Reply:
thx for the feedback I added a test for that in current PR